ACEHAL/fit.py

```diff
@@ -315,23 +315,24 @@ def do_fit(Psi, Y, B, E0s, solver, n_committee=8, basis_normalization=None, pot_
     if n_committee > 0:
         sigma = solver.sigma_
 
-        # ARD solver returns sigma only for selected features, but does not indicate which.
-        # Reconstruct full signa, hoping that coeff == 0 implies feature was removed by ARD
+        # sklearn ARDRegression solver returns sigma only for selected features, but does not explicitly
+        # indicate which ones those are.
         if sigma.shape[0] != len(c_norm):
-            # make sure sizes match
-            assert sigma.shape[0] == sum(np.nonzero(c_norm))
+             # only valid for sklearn ARDRegression
+            included_c = solver.lambda_ < solver.threshold_lambda
+            assert sigma.shape[0] == sum(included_c)
 
-            sigma = np.zeros((len(c_norm), len(c_norm)), dtype=float)
-            for (i, non_zero) in enumerate(np.nonzero(c_norm)):
-                coeff = solver.sigma_[i, i]
-                sigma[non_zero, non_zero] = coeff
+            sigma_full = np.zeros((len(c_norm), len(c_norm)), dtype=sigma.dtype)
+            inds = np.where(included_c)[0]
+            sigma_full[inds[:, None], inds] += sigma
```

Comment:
`sigma_full` is already 0, so no need for the `+=`. Also, any reason why it's not just `sigma_full[inds, inds]`?

Reply:
Also, what makes you think the `np.where(included_c)` is necessary?  Using bool arrays as masks for the indices in an `ndarray` should work.  I just checked that syntax of just `sigma_full[included_c, included_c]`, and it seemed to work OK.

Reply:
> Am I correct that now `bases/default.py` and `bases/smooth.py` are identical except for the smoothness prior? IF that's true, should there be just one option, with a hyper controlling the smoothness prior? We could extend it to different variants, too, e.g. the exponential that's being discussed in the slack.

Yes perhaps this is the right way to go, just keep the default and have options for the different smoothness priors

Reply:
For some reason I get 
```ValueError: shape mismatch: value array of shape (318,318) could not be broadcast to indexing result of shape (318,)```
When I
```sigma_full[included_c, included_c] = sigma```?

Reply:
This is why I went the option above

Reply:
Weird.  Let me test it, because I'm sure I tested the syntax I suggested (although not in this exact function) and it worked.

Reply:
No, you're right.  Maybe I only checked 1-D arrays?  Anyway, I still think we should replace the `+=` with just `=`.

The simplest way to avoid doing the `np.where` is to use `sigma_full[included_c, :][:, included_c]`, but it's only a marginal improvement, even in my opinion.

Reply:
Done